src/hooks/useNotifications.ts

```diff
@@ -0,0 +1,27 @@
+import { gql, useQuery } from '@apollo/client'
+
+const userNotificationQuery = gql`
+  query {
+    me {
+      userID
+      concreteNotifications {
+        id
+        headline
+        body
+        notification {
+          messageType
+        }
+        createdAt
+      }
+    }
+  }
+`
+
+const useNotifications = (skip?: number, take?: number) => {
+  const { data, loading, error, refetch } = useQuery(userNotificationQuery)
+  // TODO: on first request me seems to be empty, not evaluation
+  const notifications = data?.me?.concreteNotifications
+  return { data, notifications, loading, error, refetch }
```

Comment:
You should either use an effect here or do not destructure the notifications. Otherwise the notifications are not updated properly.
In addition, I wouldn't just return data, but only the notifications.
```suggestion
  const { data, loading, error, refetch } = useQuery(userNotificationQuery)
  
  const [notifications, setNotifications] = useState([]);
  
  useEffect(() => {
    if(!loading && !error) {
      setNotifications(data?.me?.concreteNotifications);
    }
  }, [loading]);
  
  return { notifications, loading, error, refetch }
```